frontend/src/components/DocumentClass.js

```diff
@@ -65,27 +72,43 @@ class DocumentClass extends Component {
     this.setState({ description: event.target.value })
   }
 
-  handleSubmit() {
-    updateDocumentClass(
+  async handleSubmit() {
+    this.props.beginLoading()
+    await updateDocumentClass(
       this.props.documentClass._id,
       this.state.name,
       this.state.description,
       this.state.files[0],
       this.state.files[0].name
     )
+    const document_classes = await getAllDocumentClasses()
+    if (document_classes) {
+      this.props.updateDocumentClasses(document_classes)
+    } else {
+      this.props.updateDocumentClasses([])
+    }
+    this.props.endLoading()
     this.uploadToggle()
   }
 
   // for if a Document Class is being submitted without an example file
   // may be removed later
-  handleSubmitNoFile() {
-    updateDocumentClass(
+  async handleSubmitNoFile() {
```

Comment:
Can this function be consolidated with the handleSubmit function? It seems like the only difference here is that some of the fields is null, which can be handled in the function above.

Reply:
Fixed